src/irr/asset/IGLSLCompiler.cpp

```diff
@@ -27,16 +27,22 @@ IGLSLCompiler::IGLSLCompiler(io::IFileSystem* _fs) :
     builtinLdr->drop();
 }
 
-ICPUShader* IGLSLCompiler::createSPIRVFromGLSL(const char* _glslCode, E_SHADER_STAGE _stage, const char* _entryPoint, const char* _compilationId, std::string* _outAssembly) const
+ICPUShader* IGLSLCompiler::createSPIRVFromGLSL(const char* _glslCode, E_SHADER_STAGE _stage, const char* _entryPoint, const char* _compilationId, bool _genDebugInfo, std::string* _outAssembly) const
 {
     //shaderc requires entry point to be "main" in GLSL
     if (strcmp(_entryPoint, "main") != 0)
         return nullptr;
 
+#ifdef _IRR_COMPILE_WITH_OPENGL_
```

Comment:
`asset` namespace violation (cannot depend on `video`)... just analyse whether the SPIR-V had debug info and `assert` that it did.

Also doesn't the requirement that debug info is available make it impossible to use .spv pulled off the web?

Reply:
>Also doesn't the requirement that debug info is available make it impossible to use .spv pulled off the web?

yes, I wrote it in the docs in iglslcompiler, but there's no other way to set push constants in opengl backend

Reply:
yes, you can just make the production of debug data default to true + assert or fail to produce the opengl gpu shader if introspection tells us there are push constants but no debug data to know their names.

Reply:
resolved